rllib/agents/dreamer/dreamer.py

```diff
@@ -250,3 +278,44 @@ def execution_plan(workers, config, **kwargs):
             )
         )
         return rollouts
+
+    @override(Trainer)
+    def training_iteration(self) -> ResultDict:
+        local_worker = self.workers.local_worker()
+
+        # Number of sub-iterations for Dreamer
+        dreamer_train_iters = self.config["dreamer_train_iters"]
+        batch_size = self.config["batch_size"]
+        action_repeat = self.config["action_repeat"]
+
+        # Collect SampleBatches from rollout workers.
+        batch = synchronous_parallel_sample(worker_set=self.workers)
+
+        fetches = {}
+
+        # Dreamer training loop.
+        # Run multiple sub-iterations for each training iteration.
+        for n in range(dreamer_train_iters):
+            print(f"sub-iteration={n}/{dreamer_train_iters}")
+            batch = self.local_replay_buffer.sample(batch_size)
+            fetches = local_worker.learn_on_batch(batch)
+
+        if fetches:
+            # Custom Logging
+            policy_fetches = fetches[DEFAULT_POLICY_ID]["learner_stats"]
+            if "log_gif" in policy_fetches:
+                gif = policy_fetches["log_gif"]
+                policy_fetches["log_gif"] = self._postprocess_gif(gif)
+
+        self._counters[STEPS_SAMPLED_COUNTER] = self.local_replay_buffer.timesteps
+        self._counters[STEPS_SAMPLED_COUNTER] *= action_repeat
+
+        # TODO: Need to update res somewhere?
```

Comment:
Good question. Metrics of the rollouts are now computed outside of the training_iteration method by Trainer (in a generic way). This may be a problem here for Dreamer that we need to solve. ...

Reply:
Actually, this looks all like standard stuff here. Just counters. So I don't think we have to do anything here (just remove these commented-out lines). Maybe confirm the counter stats are there after(!) the `DreamerTrainer.train()` call, as Trainer should automatically append them to the stats after the call to `training_iteration()`.